File: 0238-product-of-array-except-self/0238-product-of-array-except-self.c
#include <stdlib.h>

/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* productExceptSelf(int* nums, int numsSize, int* returnSize) {
    int* answer = (int*)malloc(sizeof(int) * numsSize);
    *returnSize = numsSize;

    // 1. answer[i]에 nums[i] 왼쪽에 있는 모든 원소의 곱 저장
    int leftProduct = 1;

    for (int i = 0; i < numsSize; i++) {
        answer[i] = leftProduct;
        leftProduct *= nums[i];
    }

    // 2. 오른쪽 곱을 누적하면서 answer[i]에 곱해주기
    int rightProduct = 1;

    for (int i = numsSize - 1; i >= 0; i--) {
        answer[i] *= rightProduct;
        rightProduct *= nums[i];
    }

    return answer;
}

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
from typing import List

class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        answer = [1] * n

        # 1. answer[i]에 nums[i] 왼쪽에 있는 모든 원소의 곱 저장
        left_product = 1
        for i in range(n):
            answer[i] = left_product
            left_product *= nums[i]

        # 2. 오른쪽 곱을 누적하면서 answer[i]에 곱해주기
        right_product = 1
        for i in range(n - 1, -1, -1): # range(start, stop, step):
            answer[i] *= right_product
            right_product *= nums[i]

        return answer